centaur/src/main/scala/centaur/test/formulas/TestFormulas.scala

```diff
@@ -337,6 +357,15 @@ object TestFormulas extends StrictLogging {
       case _ => Test.invalidTestDefinition("Configuration not supported by PapiUpgradeTest", workflowDefinition)
     }
 
+  def runSuccessfulWorkflowAndVerifyCost(
```

Comment:
Did you consider updating the existing test format to optionally check cost (if it's provided in the test config)?